### PR TITLE
Refactor resty client retry conditions

### PR DESCRIPTION
**Pull request by zliang-akamai**
## 📝 Description

This is to make retry conditions code more readable and clean.

## ✔️ How to Test

We can try them via instance and image automated tests:

- `make testacc PKG_NAME="linode/image"`
- `make testacc PKG_NAME="linode/images"`
- `make testacc PKG_NAME="linode/instance"`

### CODE REVIEW

**Review by jcallahan-akamai — COMMENTED** [review]
Thanks for this; it's much cleaner.

Do the functions in `/helper/retry.go` actually need to be exported now? (except for `ApplyAllConditions`)

**Comment by zliang-akamai** [comment]
Hey @jcallahan-akamai, thanks for the comment. They are not currently exported to anywhere within this package.

**Review by lgarber-akamai — APPROVED** [review]
This is a great readability improvement, excellent work!

File: linode/helper/retry.go [hunk 1]
```diff
@@ -69,6 +48,12 @@ func ImageUpload500Retry() func(response *resty.Response, err error) bool {
 		}
 
 		// Check whether the string matches
-		return ImageUpload.MatchString(requestURL.Path)
+		return pathPattern.MatchString(requestURL.Path)
 	}
 }
+
+func ApplyAllConditions(client *linodego.Client) {
```
Comment (yec-akamai) [inline linode/helper/retry.go:55]: nit: Probably more clear to add `retry` in the function names, i.e. `GenericRetryCondition`, `ApplyAllRetryConditions`?

**Review by yec-akamai — APPROVED** [review]
Nice improvement! Just an optional naming suggestion.

File: linode/helper/retry.go (hunk 1 above)
Comment (zliang-akamai) [reply linode/helper/retry.go]: Thanks for the suggestion! I just added it in the latest commit.

**Pull request merged**